lib/apollo-exception.filter.ts

```diff
@@ -0,0 +1,39 @@
+import { Catch, ArgumentsHost, HttpException } from '@nestjs/common';
```

Comment:
I wondering if this class is needed. This feels more like an example on how to implement exception filters for GraphQL. Most people will not use this directly  since custom logic such as logging etc might be required depending on use case.  Maybe as a base class similar to `BaseExceptionFilter`.

Another approach is to define a helper to convert HTTPExceptions to Apollo Errors.

```javascript
import { ApolloError, AuthenticationError, ForbiddenError } from 'apollo-server-errors'
import { HttpException } from '@nestjs/common'

const apolloPredefinedExceptions: Record<number, typeof ApolloError> = {
  401: AuthenticationError,
  403: ForbiddenError
}

export function convertErrorToApolloError(exception: HttpException): ApolloError {
  let error: ApolloError
  if (exception.getStatus() in apolloPredefinedExceptions) {
    error = new apolloPredefinedExceptions[exception.getStatus()](
      exception.message
    )
  } else {
    error = new ApolloError(
      exception.message,
      exception.getStatus().toString()
    )
  }

  error.stack = exception.stack
  error.extensions['response'] = exception.getResponse()
  return error
}
``` 

Then this helper can be used in any custom `ExceptionFilter` implementing `GqlExceptionFilter` like below.

```javascript
@Catch(HttpException)
export class MyCustomExceptionFilter implements GqlExceptionFilter {
  catch(exception: HttpException, host: ArgumentsHost) {
   const apolloError: ApolloError = convertErrorToApolloError(exception)

    this.logger.error(apolloError)
    return apolloError;
  }
}
```

Reply:
I think this class should be as part of the infra so you get this behavior "out of the box" without needing to worry about the internals of Nest exceptions vs. Apollo exceptions. And this is indeed the plan (see @kamilmysliwiec comment above), but I'm changing its name to be `BaseGqlExceptionFilter` for now so it'll be clear that you should inherit from it in GraphQL environments.